In [1]:
import requests
import pandas as pd
from bs4 import BeautifulSoup

In [1]:
def tmp():
    pass            

**pass**

- 함수 골격만 먼저 작성할 때

- 아직 로직을 작성하지 않은 임시 자리표시자

- 에러 없이 빈 블록을 유지할 때

In [5]:
def get_lyrics(a,b):
    return a + b

In [6]:
get_lyrics(10, 20)

30

In [ ]:
def get_lyrics(a, b=10):        # 기본값은 뒤에서부터 
    return a + b

In [8]:
get_lyrics(100)

110

In [ ]:
abcd = 99
def get_lyrics(a:int, b:int) -> int :       # 함수는 int를 반환할 것으로 예상
    abcd = 10
    print(abcd)         # 지역변수 호출
    return a + b
get_lyrics(10, 20)
print(abcd)             # 전역변수 호출

10
99


In [ ]:
abcd = 99
def get_lyrics(a:int, b:int) -> int :
    global ff           # 내부에서 전역 변수에 접근하거나 값을 변경할 때 사용
    ff = 10
    print(abcd)         
    return a + b
get_lyrics(10, 20)
print(ff)

99
10


In [8]:
def get_lyrics():
    return 10, 20

In [9]:
a = get_lyrics()

In [10]:
a

(10, 20)

In [11]:
a, b = get_lyrics()

In [12]:
a

10

---

In [54]:
import requests
url = "https://www.melon.com/song/detail.htm?songId=600354699"

In [5]:
requests.get(url)

<Response [406]>

**HTTP 406 Not Acceptable** : 클라이언트가 요청한 콘텐츠 형식을 서버가 제공할 수 없음

In [6]:
head = {"user-agent" :
        "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/142.0.0.0 Safari/537.36"
        }

In [55]:
lyrics = requests.get(url, headers=head).text

**div**

- 요소들을 묶어서 그룹화

- 레이아웃 배치(행, 열 구조) 구성

- CSS 스타일 적용의 기본 단위

- 자바스크립트로 특정 영역을 제어할 때 기준 역할

In [57]:
from bs4 import BeautifulSoup
bs = BeautifulSoup(lyrics)

In [58]:
# 특정 div를 HTML에서 찾기
bs.find("div", id="d_video_summary")        

# div 내부의 텍스트를 줄바꿈까지 유지하여 출력
print(BeautifulSoup(str(bs.find("div", id="d_video_summary")).replace("<br/>", "\n")).text)


							This is a hot spot
숨 쉬듯 찾는 네 밥상
단골이 된 넌 fall in love
참을 수 없는 맛

우린 in the kitchen
Cookin’ it up what you're craving
나의 입맛대로 saucin’
And now the world's gone mad

You don’t need to think too much
씹고 뜯고 맛보고 즐기다 가면 돼
Yeah yeah yeah
깊이 스며들어 in your mouth
Guilty pleasure never killed nobody 드셔봐

이빨 사이 낀 SPAGHETTI
빼고 싶니 bon appétit
그냥 포기해 어차피
Eat it up eat it eat it up

머릿속 낀 SSERAFIM
Bad bitch in between your teeth
그냥 포기해 어차피
Eat it up eat it eat it up

Eat it up eat it eat it 
Eat it up eat it up
Eat it up eat it eat it 
Eat it up eat it eat it up 

This is the sweet spot
Heat up the scene, it's that big shot
Chef's choice, 널 위한 킥이야
You lost in the sauce, no turnin' back

더 취할 거야 밤새, 기울여 절로
난 빈 잔은 받고, 이젠 핀잔은 걸러
좋은 게 좋은 거지 좋다는 게, 왜 뭐?
느껴봐 어서, 꺾지 말고, 벌컥

Snap it up like a Getty
내 김치 포즈는 여전히 priceless
Whatever I’m cookin’, 스타같이
어떤 코스든 원하면 말아주는 쉐키

내가 네 취향 드라마 주인공이고
또 숨.듣.명이고, 고칼로리 Hershey 초코
모든 줏대를 홀리는 타로, 넌 음미해 나로
자 이제 까에서 빠로

Yeah this is a hot spot
Don’t give a fuck 네가 뭐라던
씹어보셔 맛이 

In [59]:
# strip() : 문자열 앞뒤에 있는 공백·개행문자·탭을 제거

lyrics_txt = BeautifulSoup(str(bs.find("div", id="d_video_summary")).replace("<br/>", "\n")).text.strip()

In [60]:
lyrics_txt

"This is a hot spot\n숨 쉬듯 찾는 네 밥상\n단골이 된 넌 fall in love\n참을 수 없는 맛\n\n우린 in the kitchen\nCookin’ it up what you're craving\n나의 입맛대로 saucin’\nAnd now the world's gone mad\n\nYou don’t need to think too much\n씹고 뜯고 맛보고 즐기다 가면 돼\nYeah yeah yeah\n깊이 스며들어 in your mouth\nGuilty pleasure never killed nobody 드셔봐\n\n이빨 사이 낀 SPAGHETTI\n빼고 싶니 bon appétit\n그냥 포기해 어차피\nEat it up eat it eat it up\n\n머릿속 낀 SSERAFIM\nBad bitch in between your teeth\n그냥 포기해 어차피\nEat it up eat it eat it up\n\nEat it up eat it eat it \nEat it up eat it up\nEat it up eat it eat it \nEat it up eat it eat it up \n\nThis is the sweet spot\nHeat up the scene, it's that big shot\nChef's choice, 널 위한 킥이야\nYou lost in the sauce, no turnin' back\n\n더 취할 거야 밤새, 기울여 절로\n난 빈 잔은 받고, 이젠 핀잔은 걸러\n좋은 게 좋은 거지 좋다는 게, 왜 뭐?\n느껴봐 어서, 꺾지 말고, 벌컥\n\nSnap it up like a Getty\n내 김치 포즈는 여전히 priceless\nWhatever I’m cookin’, 스타같이\n어떤 코스든 원하면 말아주는 쉐키\n\n내가 네 취향 드라마 주인공이고\n또 숨.듣.명이고, 고칼로리 Hershey 초코\n모든 줏대를 홀리는 타로, 넌 음미해 나로\n자 이제 까에서 빠로\n\nYeah this is 

In [61]:
bs.find("div", class_="entry")

<div class="entry">
<div class="info">
<div class="song_name">
<strong class="none">곡명</strong>
							
							
							
							
							SPAGHETTI (feat. j-hope of BTS)
						</div>
<div class="artist">
<a class="artist_name" href="javascript:melon.link.goArtistDetail('3092950')" title="LE SSERAFIM (르세라핌)"><span>LE SSERAFIM (르세라핌)</span><span class="thumb_atist"><img height="52" onerror="WEBPOCIMG.defaultArtistImg(this);" src="https://cdnimg.melon.co.kr/cm2/artistcrop/images/030/92/950/3092950_20251021140219.jpg" width="52"/></span></a>, <a class="artist_name" href="javascript:melon.link.goArtistDetail('725984')" title="j-hope"><span>j-hope</span><span class="thumb_atist"><img height="52" onerror="WEBPOCIMG.defaultArtistImg(this);" src="https://cdnimg.melon.co.kr/cm2/artistcrop/images/007/25/984/725984_20250613110810.jpg" width="52"/></span></a>
</div>
</div>
<div class="meta">
<dl class="list">
<dt>앨범</dt>
<dd><a href="javascript:melon.link.goAlbumDetail('12175901');">SPAGHETTI</a></dd>

In [62]:
bs.find("div",{ "class":"entry"})
bs.find("div", class_="entry")

<div class="entry">
<div class="info">
<div class="song_name">
<strong class="none">곡명</strong>
							
							
							
							
							SPAGHETTI (feat. j-hope of BTS)
						</div>
<div class="artist">
<a class="artist_name" href="javascript:melon.link.goArtistDetail('3092950')" title="LE SSERAFIM (르세라핌)"><span>LE SSERAFIM (르세라핌)</span><span class="thumb_atist"><img height="52" onerror="WEBPOCIMG.defaultArtistImg(this);" src="https://cdnimg.melon.co.kr/cm2/artistcrop/images/030/92/950/3092950_20251021140219.jpg" width="52"/></span></a>, <a class="artist_name" href="javascript:melon.link.goArtistDetail('725984')" title="j-hope"><span>j-hope</span><span class="thumb_atist"><img height="52" onerror="WEBPOCIMG.defaultArtistImg(this);" src="https://cdnimg.melon.co.kr/cm2/artistcrop/images/007/25/984/725984_20250613110810.jpg" width="52"/></span></a>
</div>
</div>
<div class="meta">
<dl class="list">
<dt>앨범</dt>
<dd><a href="javascript:melon.link.goAlbumDetail('12175901');">SPAGHETTI</a></dd>

In [63]:
bs.find("div", class_="entry").find("div", class_="song_name").text.strip().replace("곡명", "").strip()

'SPAGHETTI (feat. j-hope of BTS)'

In [64]:
bs.find("div", class_="entry").find('a')['title']

'LE SSERAFIM (르세라핌)'

In [ ]:
def get_lyrics(song_id:str) -> dict:
    """
    song_id : 노래 ID
    return : {'title' : '노래제목', 'artist': '가수', 'lyrics': '가사'}
    """

In [ ]:
help(get_lyrics)

Help on function get_lyrics in module __main__:

get_lyrics(song_id: str) -> dict
    song_id : 노래 ID
    return : {'title' : '노래제목', 'artist': '가수', 'lyrics': '가사'}



In [80]:
def get_lyrics(song_id:str) -> dict:
    """
    song_id : 노래 ID
    return : {'title' : '노래제목', 'artist': '가수', 'lyrics': '가사'}
    """
    url = f"https://www.melon.com/song/detail.htm?songId={song_id}"
    head = {"user-agent" : "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/142.0.0.0 Safari/537.36 Edg/142.0.0.0"}
    lyrics = BeautifulSoup(requests.get(url, headers=head).text)

    title = bs.find("div", class_="entry").find("div", class_="song_name").text.strip().replace("곡명", "").strip()
    artist = bs.find("div", class_="entry").find('a')['title']
    lyrics_txt = BeautifulSoup(str(bs.find("div", id="d_video_summary")).replace("<br/>", "\n")).text.strip()
    
    return {'title' : title, 'artist': artist, 'lyrics': lyrics_txt }


In [81]:
get_lyrics("600354699")

{'title': 'SPAGHETTI (feat. j-hope of BTS)',
 'artist': 'LE SSERAFIM (르세라핌)',
 'lyrics': "This is a hot spot\n숨 쉬듯 찾는 네 밥상\n단골이 된 넌 fall in love\n참을 수 없는 맛\n\n우린 in the kitchen\nCookin’ it up what you're craving\n나의 입맛대로 saucin’\nAnd now the world's gone mad\n\nYou don’t need to think too much\n씹고 뜯고 맛보고 즐기다 가면 돼\nYeah yeah yeah\n깊이 스며들어 in your mouth\nGuilty pleasure never killed nobody 드셔봐\n\n이빨 사이 낀 SPAGHETTI\n빼고 싶니 bon appétit\n그냥 포기해 어차피\nEat it up eat it eat it up\n\n머릿속 낀 SSERAFIM\nBad bitch in between your teeth\n그냥 포기해 어차피\nEat it up eat it eat it up\n\nEat it up eat it eat it \nEat it up eat it up\nEat it up eat it eat it \nEat it up eat it eat it up \n\nThis is the sweet spot\nHeat up the scene, it's that big shot\nChef's choice, 널 위한 킥이야\nYou lost in the sauce, no turnin' back\n\n더 취할 거야 밤새, 기울여 절로\n난 빈 잔은 받고, 이젠 핀잔은 걸러\n좋은 게 좋은 거지 좋다는 게, 왜 뭐?\n느껴봐 어서, 꺾지 말고, 벌컥\n\nSnap it up like a Getty\n내 김치 포즈는 여전히 priceless\nWhatever I’m cookin’, 스타같이\n어떤 코스든 원하면 말아주는 쉐키\n\n내가 네 취향 드라마